package/gluon-core/luasrc/usr/lib/lua/gluon/util.lua

```diff
@@ -188,4 +190,78 @@ function M.log(message, verbose)
 	posix_syslog.syslog(posix_syslog.LOG_INFO, message)
 end
 
+-- Blindly close the file descriptors in a given table.
+local function close_fds(descriptors)
+	for _, fd in pairs(descriptors) do
+		posix_unistd.close(fd)
+	end
+end
+
+M.subprocess = {}
+
+M.subprocess.DEVNULL = -1
+M.subprocess.PIPE = 1
+
+-- Execute a program found using command PATH search, like the shell.
+-- Return the pid, as well as the I/O streams as pipes or nil on error.
+function M.subprocess.popen(path, argt, options)
```

Comment:
How about adding `argt = argt or {}` somewhere? This would allow to call the function with `argt == nil` - in particular, the `argt` and `options` arguments can be omitted completely when they are not needed.

Reply:
resolved in d513d001f4ea9614a8dd36bec4f572148583ae71